src/setting_definitions/array.py

```diff
@@ -0,0 +1,12 @@
+from .definition import SettingDefinition
+
+
+class ArraySettingDefinition(SettingDefinition):
```

Comment:
And this can come out now, right? (along with the corresponding import).  Otherwise this looks cool - it's a really small set of changes for something that is going to make the images way nicer to deal with